tests/utils/crypt.js

```diff
@@ -0,0 +1,207 @@
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/**
+ * tests/utils/cryptUtils.js
+ */
+'use strict'; // eslint-disable-line strict
+const expect = require('chai').expect;
+const dbConstants = require('../../db/constants');
+const generatorUtils = require('../../utils/cryptUtils');
+const tu = require('../testUtils');
+const GlobalConfig = tu.db.GlobalConfig;
+
+describe('tests/utils/cryptUtils.js >', () => {
+  const secretKey = 'mySecretKey';
+  const algorithm = 'aes-256-cbc';
+
+  beforeEach((done) => {
+    GlobalConfig.create({
+      key: dbConstants.SGEncryptionKey,
+      value: secretKey,
+    })
+    .then(() => GlobalConfig.create({
+      key: dbConstants.SGEncryptionAlgorithm,
+      value: algorithm,
+    }))
+    .then(() => done())
+    .catch(done);
+  });
+
+  afterEach((done) => {
+    GlobalConfig.destroy({ truncate: true, force: true })
+    .then(() => done())
+    .catch(done);
+  });
+
+  describe('encrypt/decrypt', () => {
+    it('encrypt a text and decrypt it again to make sure the decrypted ' +
+      'text matches the original text', (done) => {
+      const data = 'My data';
+      const encryptedData = generatorUtils.encrypt(data, secretKey, algorithm);
+      const decryptedData = generatorUtils.decrypt(encryptedData,
+        secretKey, algorithm);
+      expect(decryptedData).to.equal(data);
+      done();
+    });
+  });
+
+  describe('getSGEncryptionKeyAndAlgorithm', () => {
+    it('object returned must have secretKey and algorithim as keys', (done) => {
+      generatorUtils.getSGEncryptionKeyAndAlgorithm(GlobalConfig)
+      .then((gc) => {
+        expect(gc.secretKey).to.equal(secretKey);
+        expect(gc.algorithm).to.equal(algorithm);
+        done();
+      }).catch(done);
+    });
+  });
+
+  describe('encryptSGContextValues', () => {
+    it('when encrypted is set to true in SGT, the related SG values ' +
+      'should be encrypted', (done) => {
+      const looksLikeSGT = {
+        name: 'My_TEST_SGT',
+        version: '1.0.0',
+        contextDefinition: {
+          password: {
+            encrypted: true,
+          },
+          secretInformation: {
+            encrypted: true,
+          },
+          otherNonSecretInformation: {
+            encrypted: false,
+          },
+        },
+      };
+
+      const password = 'superlongandsupersecretpassword';
+      const secretInformation = 'asecretthatyoushouldnotknow';
+      const otherNonSecretInformation = 'nonsecreInformation';
+      const looksLikeSG = {
+        name: 'MY_TEST_SG',
+        version: '1.0.0',
+        context: {
+          password,
+          secretInformation,
+          otherNonSecretInformation,
+        },
+      };
+      generatorUtils.encryptSGContextValues(GlobalConfig,
+        looksLikeSG, looksLikeSGT)
+      .then((sg) => {
+        expect(sg).to.not.equal(undefined);
+        expect(sg.context.otherNonSecretInformation)
+          .equal(otherNonSecretInformation);
```

Comment:
expect(sg.context.secretInformation).to.not.equal(secretInformation); please.

Reply:
if they are equal the decryption will fail. line 107 proves that they were not equal.

Reply:
I was saying *not.equal*, but nevermind, I tested this on local and it works. I can add it in my PR. I am fine merging the PR. Thanks!!

Reply:
Yes, "not.equal". Saying something is not equal to something does not help much and in this case, it does not tell you that the value was encrypted. 
According to the "chai expects"  library it is not a recommended way 
`expect(2).to.not.equal(1); // Not recommended` 
We have done in in quite a few places in the tests but it is something we should avoid. 
I tend to use not.equals usually before referencing the object's properties.

Reply:
Right. I absolutely agree. We should NOT prefer "not.equal" over "equal", but in cases we cannot test "equal", then I guess testing atleast "not.equal" is better. 
Like in this scenario, lets say encrypt/decrypt funcs are not doing anything and returning the object as it is, then the test will still pass.

Reply:
There are other tests that test the encrypt/decrypt functions, those tests will fail.

Reply:
I just commented out the encrypt and decrypt functions and return just data. All the tests pass in tests/utils/crypt.js

Reply:
if you returned a random string instead of the data then having the not.equal test would pass. I do not understand what you are trying to prove here.

Reply:
Nevermind, nothing to prove, just a remote discussion on whether adding another line in test would make it better or not. Hangouts would have been better :)